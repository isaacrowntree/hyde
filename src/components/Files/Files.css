.list-group {
  width: 16%;
  position: absolute;
  z-index: 100;
}
