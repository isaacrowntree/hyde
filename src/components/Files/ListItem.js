import React, { Component } from 'react';

class ListItem extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.onItemClick(this.props.value);
  }

  format(val) {
    return val.split('\\').slice(2).join('\\');
  }

  render() {
    return (
      <li className="list-group-item" onClick={this.handleClick}>
        {this.format(this.props.value)}
      </li>
    );
  }
}

export default ListItem;
