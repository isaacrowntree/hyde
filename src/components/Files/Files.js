import React, { Component } from 'react';
import axios from 'axios';

import ListItem from './ListItem.js';
import Editor from './../Editor/Editor.js';

const config = require('./../../../server/config');

class Files extends Component {
  constructor(props) {
    super(props);

    this.state = {
      files: [],
      file: '',
    };
    this.edit = this.edit.bind(this);
  }

  componentDidMount() {
    axios.get(`http://localhost:${config.port}/files`).then(res => {
      this.setState({ files: res.data });
    });
  }

  edit(value) {
    this.setState({ file: value });
  }

  render() {
    var rows = [];
    var counter = 0;
    for (var value of this.state.files) {
      counter++;
      if (value.match(/\.md$/)) {
        rows.push(<ListItem onItemClick={this.edit} key={counter} value={value} />);
      }
    }
    return (
      <div>
        { rows }
        <Editor file={this.state.file} />
      </div>
    );
  }
}

export default Files;
