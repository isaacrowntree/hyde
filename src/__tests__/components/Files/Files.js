import React from 'react';
import renderer from 'react-test-renderer';
import Files from '../../../components/Files/Files';

test('Files renders correctly', () => {
  const tree = renderer
    .create(<Files />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
