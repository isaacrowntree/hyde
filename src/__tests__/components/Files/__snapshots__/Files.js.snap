// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Files renders correctly 1`] = `
<div>
  <div
    className="list-group"
  />
  <div
    className="row"
  >
    <div
      className="col-md-5 offset-md-2"
    >
      <textarea
        className="Editor"
        onChange={[Function]}
        value=""
      />
      <br />
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Publish
      </button>
    </div>
    <div
      className="col-md-5"
    >
      <div
        className="interweave"
      />
    </div>
  </div>
</div>
`;
